Updated to latest gems and tested with cassandra 0.7

cirode opened the pull request:
We are using cassandra object in one of our projects having decided that updating this gem was easier and better than creating a model base ourselves. The following changes have been added
1. Vendorised cassandra 0.7 (and added a developer section to the README) to allow developers to run the tests without having cassandra installed locally
2. Updated the gems (and gemspec) to the latest versions
3. Fixed all the current tests to cope with the changes in gems

If you are amenable to it, we will be added support for cassandra/0.8  and other small enhancements as we need them for this project and these will form additional pull requests

Commit:
Replaced returning() with tap() to avoid deprecation warning message.

Commit:
got the tests running
vendorised cassandra to get the tests running
only 12 failures now

cirode committed:
fixed two bugs to do with old style assertion mapping

Commit:
Fix: get_range() was expected to return just keys, while it is returning whole objects.

cirode committed:
removing the data from the vendored cassandra

cirode committed:
removed the vendored cassandra within vendor cassandra

cirode committed:
ignoring the cassandra data
fixed an issue with the index cleanup. Seems someone got confused with the amount of variables marked as "key"..

cirode committed:
modified the cassndra log to not create many subdirectories

cirode committed:
removed the cassandra logs as well

cirode committed:
added some development notes on testing

Commit:
Added clear! method to clear non-reversed associations.

cirode committed:
updated the gems to the latest and reflected in gemspec

Turns out tzinfo is no longer a dependency of active_support and so needed to be manually added

Commit:
Added count() method for base class, to return the number of items stored in Cassandra.
E.g.: Customer.count # returns the number of Customers items

Commit:
Fix: the :boolean attribute type was not supported.

fuentesjr commented:
+1 
Please approve.

Commit:
The :natural key now supports methods in the list of attributes.
E.g.: key :natural, :attributes => [:an_attribute, :a_method]

matthuhiggins commented:
We also have a maintained fork at https://github.com/gotime/cassandra_object. The history that it was forked was lost because it started as a private repo.

Commit:
Updated cassandra gem to latest version (0.12.1).

Commit:
Made sure we always call get_range with a :key_count parameter. Otherwise, when tombstones are in Cassandra the all() method may return zero results regardless of the number of valid results it should be returning.
Fixed get_range parameters (were referring to some very old version of the cassandra gem).

Commit:
Updated cassandra gem dependency to version 0.12.0.

Commit:
doesn't require :limit to be passed in order to iterate through all objects and associations

Commit:
Updated activesupport and activemodel dependencies to 3.1.x.

Commit:
Added .rvmrc to gitignore.
